geopandas/sindex.py

```diff
@@ -32,6 +32,20 @@ def _get_sindex_class():
     )
 
 
+class UnknownPredicateError(ValueError):
+    def __init__(self, predicate, valid_predicates=None, message=None):
+        valid_predicates = valid_predicates or set()
+        message = message or "Predicate {} is not valid; must be one of {}".format(
+            predicate,
+            valid_predicates,
+        )
+        super().__init__(message)
+
+
+class UnknownGeometryType(ValueError):
+    pass
```

Comment:
This is doing nothing at the time since AFAIK both implementations handle all geometry types.

Reply:
Then there's no need to add it now, is it? Plus it is the same as above. `Polygon` would not be unknown, but not implemented or not supported.